mastodon.go

```diff
@@ -23,14 +24,30 @@ type Client struct {
 	config *Config
 }
 
-func (c *Client) doAPI(method string, uri string, params url.Values, res interface{}) error {
+func httpDo(ctx context.Context, req *http.Request, f func(*http.Response, error) error) error {
+	tr := &http.Transport{}
+	client := &http.Client{Transport: tr}
+	c := make(chan error, 1)
+	go func() {
+		c <- f(client.Do(req))
+	}()
+	select {
+	case <-ctx.Done():
+		tr.CancelRequest(req)
```

Comment:
This function is deprecated, but is it intended?

https://golang.org/pkg/net/http/#Transport.CancelRequest
> Deprecated: Use Request.WithContext to create a request with a cancelable context instead.